sdcm/coredump.py

```diff
@@ -103,11 +105,25 @@ def run(self):
         """
         Keep reporting new coredumps found, every 30 seconds.
         """
+        exceptions_count = 0
+        last_exc = None
         while not self.termination_event.wait(self.lookup_period) or self.in_progress:
             try:
                 self.main_cycle_body()
+                exceptions_count = 0
             except Exception as exc:
                 self.log.error(f"Following error occurred: {str(exc)}")
+                exceptions_count += 1
+                last_exc = exc
+
+            # If exceptions will happen MAX_COREDUMP_THREAD_EXCEPTIONS times - fail the thread and don't run it forever
+            if exceptions_count == self.max_coredump_thread_exceptions:
+                self.main_cycle_body_reach_exception_limit(last_exc)
+
+    # The function is added for unit test ability to recognize that exception was raised
+    @staticmethod
+    def main_cycle_body_reach_exception_limit(exc):
+        raise exc
```

Comment:
What about having attrubute `self.exception` for storing raised exception? It'll simplify code a little bit and in result we don't need to do something special for unit tests too:

```python
def __init__(self):
    ...
    self.exception = None
    ...

def run(self):
    exceptions_count = 0
    while ...:
        try:
            self.main_cycle_body()
            exceptions_count = 0
        except Exception as exc:
            self.log.error("...: %s", exc)
            exceptions_count += 1
            if exceptions_count == self.max_coredump_thread_exceptions:
                self.exception = exc
                raise
```

And in unit tests data you can replace list `raised_exceptions` with `exception`.

Reply:
changed